lib/_http_outgoing.js

```diff
@@ -296,19 +296,16 @@ function _storeHeader(firstLine, headers) {
     header: firstLine
   };
 
-  var key;
-  if (headers === this[outHeadersKey]) {
-    for (key in headers) {
-      const entry = headers[key];
+  if (headers && headers === this[outHeadersKey]) {
```

Comment:
Object.values can't handle null and undefined, so I add the check.

Reply:
If we're going to add this kind of check, we may as well wrap it for all the conditions to avoid additional checks when `headers` is falsey. For example:


```js
if (headers) {
  if (headers === this[outHeadersKey]) {
    // ...
  } else if (Array.isArray(headers)) {
    // ...
  } else {
    // ...
  }
}
```